speaker/bgp_controller.go

```diff
@@ -394,13 +417,29 @@ func (c *bgpController) SetNode(l log.Logger, node *v1.Node) error {
 	if c.myNode != node.Name {
 		return nil
 	}
+	level.Debug(l).Log("event", "nodeChange", "msg", "start")
+	defer level.Debug(l).Log("event", "nodeChange", "msg", "end")
+
+	if v := node.Spec.Unschedulable; v != c.unschedulable {
```

Comment:
we shouldn't make logic on the previous value as it's going to make the controller weak. Let' shook in the same place we have the `k8snodes.IsNetworkUnavailable` checks so the controller is level driven and not edge driven.
This will also make the clearAds not necessary.

Reply:
Can you expand more on this comment? I think I miss some logic:
the `IsNetworkUnavailable` is being within the `c.handleService` which is called by ctrl.SetBalancer`
So if understand the top down logic

`client -> k8s.Listener-> serviceChanged -> call ctrl.SetBalancer `

which executed when service changes.

How that executed when there is a node change

those are the logs when running the NetworkUnavailable
```
{"caller":"bgp_controller.go:420","event":"nodeChange","level":"debug","msg":"start","ts":"2024-07-24T13:17:27Z"}
{"caller":"bgp_controller.go:442","event":"nodeLabelsNoChange","level":"debug","msg":"Node labels same, do thing","ts":"2024-07-24T13:17:27Z"}
{"caller":"bgp_controller.go:443","event":"nodeChange","level":"debug","msg":"end","ts":"2024-07-24T13:17:27Z"}
```
thanks

Reply:
When a node status changes in a way that might make all the services not available, we reprocess the services and some of them can be revoked.

In general, if some logic is implemented in some way, it's reasonable to not add a different way to implement something similar. There are already a set of conditions that might make a node not able to announce the service.

Reply:
true, I saw the logic, thanks